### ISSUE TITLE
Fails to format file containing `for (i in start..<start + count)`

### ISSUE THREAD

**Issue by ikezedev** [pinterest/ktlint#2925]
## Expected Behavior

Should be able to format valid Kotlin for loop expression
```kt
for (i in start..<start + count) {
}
```

## Observed Behavior

Ktlint failed to parse file

`./gradlew :backend:ktlintFormat`

The debug output was a bit too much, so could not put them here.

## Steps to Reproduce

```kt
for (i in start..<start + count) {
}
```

## Your Environment

* Version of ktlint used:  "11.3.2"
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Gradle plugin
* Version of Gradle used (if applicable): 8.12
* Operating System and version:  Mac OS X 15.1.1 aarch64

**Comment by paul-dingemans** [comment]
I cannot reproduce a problem with a for loop like this. See output below:
```
 $ ktlint-1.5.0 --stdin
17:24:27.694 [main] INFO com.pinterest.ktlint.cli.internal.KtlintCommandLine -- Enable default patterns [**/*.kt, **/*.kts]
fun foo() {
    val start = 1
    val count = 10
    for (i in start..<start + count) {
    bar() // Wrongly indented
    }
}

fun bar() {}
<stdin>:5:1: Unexpected indentation (4) (should be 8) (standard:indent)

Summary error count (descending) by rule:
  standard:indent: 1
```

**Comment by paul-dingemans** [comment]
Closed due to lack of response.
